File: backend/entities/product.rb
class Product
  attr_reader :name, :type, :price

  def initialize(name:, type:)
    @name, @type = name, type
  end

  def is_physical?
    type == :physical
  end

  def is_book?
    type == :book
  end

  def is_digital?
    type == :digital
  end

  def is_membership?
    type == :membership
  end
end

File: backend/individual-assignment/src/models/Product.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Product = exports.ProductType = void 0;
var ProductType;
(function (ProductType) {
    ProductType["PHYSICAL"] = "PHYSICAL";
    ProductType["BOOK"] = "BOOK";
    ProductType["DIGITAL"] = "DIGITAL";
    ProductType["MEMBERSHIP"] = "MEMBERSHIP";
})(ProductType || (exports.ProductType = ProductType = {}));
class Product {
    constructor(name, type, price) {
        this.name = name;
        this.type = type;
        this.price = price;
    }
}
exports.Product = Product;
